hello-world-master/source/main/components/KeyboardScreen/KeyboardScreen.bs

```diff
@@ -0,0 +1,98 @@
+import "../Services/NavigationService.bs"
+
+sub init()
+    m.background = m.top.findNode("background")
+    m.keyboard = m.top.findNode("keyboard")
+    m.searchResultsLabel = m.top.findNode("searchResultsLabel")
+    m.markupGrid = m.top.findNode("markupGrid")
+
+    m.keyboard.ObserveField("text", "onKeyboardTextChange")
+
+    m.navigationService = new NavigationService(m.top)
+end sub
+
+sub showContent(event as Object)
+    m.pokemonData = event.getData()
+    m.markupGrid.content = cloneContentNode(m.pokemonData)
+
+    m.background.color = "0x808080FF"
+    m.keyboard.setFocus(true)
+end sub
+
+sub onKeyboardTextChange(event as Object)
+    query = event.getData()
+    results = CreateObject("roSGNode","ContentNode")
+    childCount = m.pokemonData.getChildCount()
+    pokemonData = cloneContentNode(m.pokemonData)
+
+    if query = ""
+        m.markupGrid.content = pokemonData
+        m.searchResultsLabel.text = "Pokemons"
+
+        return
+    end if
+
+    for i = 0 to childCount - 1
+        pokemon = pokemonData.getChild(i)
+
+        if pokemon <> invalid
+            if pokemonContainsQuery(pokemon, query) then results.appendChild(pokemon)
+        end if
+    end for
+
+    if results.getChildCount() > 0
+        m.searchResultsLabel.text = "Pokemons"
+    else
+        m.searchResultsLabel.text = "No results"
+    end if
```

Comment:
you can think of a smart way to do this population of searchResultsLabel....you have 3 ifs that define how it is populated...maybe you can set a default and only change it to "No results" with one single condition and a repopulation